pkg/metrics/iterable_series.go

```diff
@@ -13,10 +17,16 @@ type IterableSeries struct {
 // NewIterableSeries creates a new instance of *IterableSeries
 //
 // `callback` is called in the context of the sender's goroutine each time `Append` is called.
-func NewIterableSeries(callback func(*Serie), chanSize int, bufferSize int) *IterableSeries {
+func NewIterableSeries(callback func(*Serie), chanSize int, bufferSize int, hostTagProvider *HostTagProvider) *IterableSeries {
+
 	return &IterableSeries{
 		iterableMetrics: *newIterableMetric(func(value interface{}) {
-			callback(value.(*Serie))
+			serie := value.(*Serie)
+			hostTags := hostTagProvider.GetHostTags()
```

Comment:
Would it maybe make more sense to put this logic [here](https://github.com/DataDog/datadog-agent/blob/d450395b2cea9ff0e5dbd3cf40bc9f13a87ee04b/pkg/aggregator/demultiplexer.go#L103) in the demultiplexer? 

This way you don't have to pass `hostTagProvider` in here, and the `hostTagProvider` instance can live in the demultiplexer. 

This will also allow you to remove the dependency from the otlp consumer, which will hopefully resolve the go mod errors :)

Reply:
Also I think we also need to duplicate this logic for `NewIterableSketches`